docs/shortfin/llm/user/shortfin_with_sglang_frontend_language.md

```diff
@@ -0,0 +1,224 @@
+# Using `shortfin` with `sglang`
+
+This doc includes basic steps for hooking up sglang with a running Shortfin server.
+
+## Install/Start `shortfin` LLM server
+
+Follow the steps [here](https://github.com/nod-ai/SHARK-Platform/blob/main/docs/shortfin/llm/user/e2e_llama8b_mi300x.md)
+to export a model with `sharktank` and start a `shortfin` LLM server
+with that model.
+
+## Install sglang
+
+### Install sglang inside of virtual environment
+
+Currently, we have our SGLang integration located at this [forked repo](https://github.com/nod-ai/sglang).
+We can use pip to install it in the same virtual environment that we used
+to start our Shortfin LLM Server.
+
+```bash
+pip install "git+https://github.com/nod-ai/sglang.git#subdirectory=python"
+```
+
+## Getting started
+
+You can verify the installation/setup through the following examples:
+
+- [Multi-Turn Q&A Example](#multi-turn-qa-example)
+- [Fork Example](#fork-example)
+- [Benchmark Shortfin](#bench-mark-shortfin-w-sglang-bench_serving-script)
+
+## Multi-Turn Q&A example
+
+Now that we have sglang installed, we can run an example to show a multi-turn
+Q&A flow with the SGLang [Frontend Language](https://sgl-project.github.io/frontend/frontend.html):
+
+### Open python interpreter
+
+```bash
+python
+```
+
+### Run example
+
+You can copy and paste the following example into your interpreter:
+
+```python
+import sglang as sgl
+
+backend = sgl.Shortfin(base_url="http://localhost:8000") # Change base_url if running at different address
+
+sgl.set_default_backend(backend)
```

Comment:
For my own curiosity... what is actually behind this "backend"? Is this using sharktank / IREE for a llama model? Or just using shortfin for a serving engine somehow on top of a different inference engine? How do you choose between models?

Oh right, this is the _client_. The _server_ is above using `/docs/shortfin/llm/user/e2e_llama8b_mi300x.md`. I see... okay. Would it be useful to log that somewhere, or is this running at a high level of abstraction where users don't care about the details?

Just thinking through how the pieces all fit together :D

Reply:
My bad, I completely missed your review! My plan was to revisit these docs and make some edits after I found a way to get better output reliably, which I found out after implementing the integration tests (using llama3-8b-instruct, and the corresponding llama-3-instruct chat template was the key). Will push those updates in a couple of minutes.

Yeah, basically this is just a very `fancy` client. You have the Shortfin Server running in the background, and the `backend` is just a super simple class for sending the `generation` requests over to our shortfin server, when needed. It mentions at the very top that you need to walk through the `e2e_llama8b_mi300x.md` steps to start a shortfin LLM Server, but it would probably help to add some context of how the pieces fit together. Will do that

Reply:
Added brief description in `## Prerequisites/ ### Software` section